src/game/entities/controllers/PlayerController.js

```diff
@@ -28,24 +32,24 @@ class PlayerController extends EntityController {
       if (playerData !== undefined) {
         let { beta, gamma } = playerData.sensor;
 
-        if (Math.abs(beta) < MIN_SENSOR_THRESHOLD) {
+        if (beta * beta + gamma * gamma < MIN_SENSOR_THRESHOLD_SQ) {
           beta = 0;
-        }
-
-        if (Math.abs(gamma) < MIN_SENSOR_THRESHOLD) {
           gamma = 0;
         }
 
-        beta = Math.min(MAX_ANGLE, Math.max(beta, -MAX_ANGLE));
-        beta = beta / MAX_ANGLE * Math.PI * 0.5;
-        gamma = Math.min(MAX_ANGLE, Math.max(gamma, -MAX_ANGLE));
-        gamma = gamma / MAX_ANGLE * Math.PI * 0.5;
+        let length = MAX_ANGLE;
+        const sqlength = beta * beta + gamma * gamma;
+        if (sqlength > MAX_ANGLE_SQ) {
+          length = Math.sqrt(sqlength);
+        }
+        beta /= length;
+        gamma /= length;
 
-        const length = Math.sqrt(beta * beta + gamma * gamma);
+        /* length = Math.sqrt(beta * beta + gamma * gamma);
```

Comment:
Is there a point to keeping these comments? Seems like this is just the old implementation

Reply:
Comments should be removed, but they have been left to help with experimentation. The code in this specific comment had the unintended consequence of maximizing movement speed at a much lower angle than intended. Maybe this behaviour makes it feel better?

Reply:
Comments should be removed, but they have been left to help with experimentation. The code in this specific comment had the unintended consequence of maximizing movement speed at a much lower angle than intended. Maybe this behaviour makes it feel better?

Reply:
Ok, so we'll just leave it for now?

Reply:
Yes, the system needs testing and probably tweaking. After that the code can be cleaned and merged into master.